### DISCUSSION TITLE
My Credit Card Workflow

### DISCUSSION THREAD

**Discussion by ghost** [actualbudget/releases#25] [General]
I've noticed a few people getting confused with Credit Cards so I decided to post how I got it working for me. My method is different to what @jlongster showed in his dev video but I feel it works a little cleaner.

1. Create "Group" named Debt
2. Create "Category" under that group named "Credit Card 1"
3. Add a new on-budget account with a negative balance
4. On that account, amend the category of the starting balance transaction to "Credit Card 1"
5. On the budget tab, click the amount and select "Rollover Overspending"

Now that you've got it setup you'll likely be wondering how you start paying down the debt. This is how I do my workflow:

1. Budget for the debt like you would any other category. So if you've got $1000 of debt, and have $500 you want to pay off. Add 500 into the "Budgeted" column.
2. When paying off the card, do a new transaction as a transfer. So in the above example transfer $500 from your checking account to your Credit Card account. This has already been budgeted for and has reduced the balance on your budget tab.
3. That's it... it's that simple.

When you make purchases on the Credit Card, you can simply use the category of the purchase. Then you know it can be paid off as it's already budgeted for. You basically want your credit card balance to match the balance shown in "Credit Card 1" when you make a payment (pay the difference between the current balance in the account and the amount shown on the budget tab) and then book this as a transfer.

If I make a purchase that I want to carry forward as a debt. I enter the category as "Credit Card 1" which then increases the balance again and I follow the paydown workflow above to then pay it down again over time.

I'll update this guide in future with some screenshots but I hope this helps!

**Comment by LsBrns** [comment]
Very helpful - thank you for sharing!

**Comment by mookerific** [comment]
To be clear, is this an effort to replicate nYNAB's CC handling in Actualbudget?

**Comment by ghost** [comment]
Nope, its replicating YNAB4

**Comment by Keetman** [comment]
I was playing with this last night and came up with something similar, or maybe even the exact thing.

I don't have old credit card debt, but I understand the problem.  If someone has old debt on a card they are not using, they could just keep it off budget and make payments on it each month (much like a mortgage).  

The problem arises when someone has old debt and they are still continuing to use the same card that debt is on, while also paying off all spending done on the current month.  Like you said, the struggle is to keep the balance on that card the actual balance and not eat into the money that is available to budget.  When you start a budget with a positive amount in your accounts, then add debt...Actual immediately takes the positive amount and subtracts the debt.  What is left over is what you have to budget.  This is impossible if you never planned to pay off that old debt in one month!

Like I said, I think I did the same thing as you.  I made a fake file and started the process of what I did.  I wanted to add some screenshots to help people understand.  Hopefully this helps because I'm not trying to steal your workflow or thread.  

Here I added a checking account with $2000.

![image](https://user-images.githubusercontent.com/94420194/142256018-42e94f73-3699-423e-99a2-d6aece1987c7.png)


Then I added a Credit Card Account with a starting balance of zero.  I want to pretend I have $25k in debt, but I don't want to start the account with that much because it will steal the money from my available to budget cash.  After the account is established, I made a transaction for a $25000 debit tied to a category called "old debt".  This makes the credit card balance -$25000, but leaves my budget ready cash at $2000.

![image](https://user-images.githubusercontent.com/94420194/142256579-a961235c-0821-456c-a547-5028eedae825.png)


Finally, this is a picture of my budget for the starting month (November) and next month.  As you said, you need to check on overspending rollover, which I did.  Now you can see that everything works together as it's supposed to.  There is the original $2000 available for budgeting, there is a $25000 old debt that will continue to carry over until it's gone.  Finally, the balance of the credit card will always total the old debt plus any current category spending each month (that will be paid off entirely).

Interest can just be added into the account as a transaction.  Payments made for the monthly spending will just come out of the categories they were spent in and any extra payments will be budgeted and come out of the "old debt" category.

Again, hopefully this helps.  If you believe I'm hijacking I'll be glad to remove.

![image](https://user-images.githubusercontent.com/94420194/142257461-fd9060e4-cd8c-466d-8d60-ba6deba9c7e9.png)

**Reply by bpaulien** [reply]
So are you saying that there's a difference between starting the CC with balance of 0, then creating a transaction and starting with a balance of whatever's on the card?  I started with the starting balance of what's on the card, and then on the 2 cards that have "old debt",  I just budgeted the amount of the old debt to the categories that "match up with" the CC accounts. 

But I'm still not sure how I would show the payment, because it would be a transfer (in the case of the card that I'm still using) but that doesn't reduce the amount of "old debt" held in the category.  (Do I just manually "unbudget" the appropriate amount from the "old debt" category?)

Also, the other card that I have, I just realized probably SHOULD be Off budget, since I'm just paying it off, but I don't see any way of converting it to Off budget.  Guess I'll add that as a feature request.

**Reply by Deguello** [reply]
I'm doing this from memory when helping friends and I haven't double-checked the scenario, so if this is wrong let me know so I can delete it so as not to add any confusion. If you already know any of this and I didn't understand that from your post, I apologize in advance.

The difference between starting with the CC under default conditions and doing it with a matching category is that the amount owed doesn't show up in Starting Balances and affect the amount to be budgeted. When assigned to the credit card category and set to rollover, it's isolated to that category and quarantined from the budget pool.

The payment for the existing debt isn't a transfer. You budget $X in addition to current spending and then pay that $X the same as a "normal" payee. It's this budgeted amount that lowers the amount that's getting rolled over and keeps the category in sync with the account balance.

**Reply by Keetman** [reply]
@Deguello 

I had actually made the payment for the old debt as a transfer.  This seemed to keep everything in line and balanced.

If I'm following you correctly, by making the old debt payment a categorized transaction, you would also have to make a corresponding transaction in the other account.  This ends up making the accounts wonky.  

If I budget an amount in the "old debt" category, it automatically reduces the balance in the category by that amount (in my example, $1000 budgeted immediately made the "old debt" category negative $24000.  When I transfer the $1000 from checking to the credit card the credit card immediately matches the "old debt" category amount.

It's tricky to conceptualize...hopefully I didn't make it worse.  I am learning this wonderful software and I most certainly not an accountant or a developer!

**Reply by Deguello** [reply]
That's fair. Maybe it was included in the transfer, but it also had to be budgeted for against the old debt category. That seems like it clicks. Maybe I need to do what you did and create a budget from scratch and walk through it step by step to refresh my memory. It's wonky enough without my making it worse.

**Reply by Keetman** [reply]
Making that fake file really helped me and I've already went back and added 6 months of data into Actual.  I don't have outstanding debt, I just wanted to see if I could make it work on a card that I used everyday...which as I understand it is really the problem (because if you aren't using the old debt card, just take it off budget).

Anyway, making the fake file helped me understand the overall working of the software better.  Hopefully it can help someone work their debt out until a better solution is made.

**Reply by Deguello** [reply]
It looks like you're budgeting the $1000 a month against the old debt category, though, right?

**Reply by Keetman** [reply]
Exactly, did I miss something?

**Reply by Deguello** [reply]
You didn't miss anything at all. That's the part that people often miss. They make the extra payment toward the old debt, but they don't budget for it and end up thinking they have more available than they do. 

Your example looks a little like what I used to do when starting a new budget even with PIF cards. I changed the pre-YNAB debt category to income this month and budgeted it against the existing balance.

**Reply by Keetman** [reply]
Gotcha.

For what it's worth, I only see 2 problems (besides a little complexity) to what I did.  One, there is no categorized transaction under the "old debt" category.  The budgeted amount is the theoretical transaction that reduces the budgeted debt.  Two, and this is a follow up to the first problem, when you budget the money into the old debt category (in my example, $1000) there is a temporary unlink between that old debt category and the CC Balance.  They aren't balanced until the payments equal what was/is budgeted.  This is likely to happen more often, I'm not sure because I only made two transactions.  It's just important to realize that the category and the old debt balance don't have to match. 

You're tracking your old debt in the category, but you always want the CC balance to match what your financial institution says it should be.  They work together, but they aren't always necessarily in exact harmony.

I hope this helps someone, but if I did miss other issues, I'm all ears.

**Reply by Keetman** [reply]
EDIT-  I just played around with it some more.  I found that I can make a starting balance with the "old debt" amount if I change the original category to the "old debt" category.  A starting balance subtracts debt from your amount "to budget" because it is defaulted into the "starting balance" category, which is treated like "income".  It can be changed.  No big deal, but it does explain why it works and it will save a transaction because you will start out where you need to be.

**Comment by Keetman** [comment]
@bpaulien 


**_"So are you saying that there's a difference between starting the CC with balance of 0, then creating a transaction and starting with a balance of whatever's on the card?"_**

This is apparently the case.  I discovered it by accident.  The way the software is set up, a starting negative balance (such as a credit card) immediately reduces the cash available to budget.  If you start with a zero balance, then show a negative transaction, it acts like any other transaction in the categories/budget.  It subtracts from the category.  If debt is created and you want to carry it over, the red arrow/overspending rollover function allows that negative transaction to carry over forever rather than be deducted from the available to budget funds.

EDIT- I just played around with it some more. I found that I can make a starting balance with the "old debt" amount if I change the original category to the "old debt" category. A starting balance subtracts debt from your amount "to budget" because it is defaulted into the "starting balance" category, which is treated like "income". It can be changed. No big deal, but it does explain why it works and it will save a transaction because you will start out where you need to be.




**_"I started with the starting balance of what's on the card, and then on the 2 cards that have "old debt", I just budgeted the amount of the old debt to the categories that "match up with" the CC accounts."_**

If I'm understanding what you did here, the problem would be with what you have available to budget.  If you have $1000 in cash and start a credit card with $500 of old debt, your "to budget" would now by $500.  Try making the starting balance zero, then making a transaction for the amount of debt you have.  You can always "undo".  Better yet, start a fake file and play around with it.  Once it clicks you can mess with your real file.


**_"But I'm still not sure how I would show the payment, because it would be a transfer (in the case of the card that I'm still using) but that doesn't reduce the amount of "old debt" held in the category. (Do I just manually "unbudget" the appropriate amount from the "old debt" category?)"_**


I added some transactions and budgeting to my fake file.  Hopefully I can explain this, if not ask another question and I'll try again.

So in this picture I have my updated budget.  I added a $200 budget in November for groceries.  I also "spent" $200 on the credit card at the super market, making my grocery budget $0 and my "To Budget" in November $1800.  In December I budgeted $1000 for "old debt".

In December, I made a $200 payment to the card to cover last month's expenses.  I also made a $1000 transfer to the card from my checking.  I haven't added any income.  So, my credit card now has a balance of $24000 and I have a "to budget" amount of $800. (which is also my checking account balance).

This accounts for an original cash balance of $2000, an old debt balance of $25000, and two payments totaling $1200.  One was categorized from groceries and one went to "old debt".  I now have totals of $800 in cash, and $24000 in debt.  This should work out in perpetuity until the $24000 is paid off.

![image](https://user-images.githubusercontent.com/94420194/142265933-a33f646e-e739-46d3-b6a5-d046738ffbc5.png)


The next two pics just show my account ledgers to help fill in the blanks of the budget....

![image](https://user-images.githubusercontent.com/94420194/142266047-c2586b47-3be3-4ac8-88b1-2a1ff564c250.png)


![image](https://user-images.githubusercontent.com/94420194/142266115-2519a41e-1092-4f61-b80d-a43f0f068720.png)



**_"Also, the other card that I have, I just realized probably SHOULD be Off budget, since I'm just paying it off, but I don't see any way of converting it to Off budget. Guess I'll add that as a feature request."_**


I'm not sure of a way to convert it, my guess is you can't because the editable features of accounts after the fact seems to be spartan.  I believe it's worth making a new account and terminating the old one because keeping old debt in unused accounts off budget is just more simple.

**Reply by bpaulien** [reply]
Thank you for the in depth reply.  I kinda sorta understand it, and I'll create a new budget tonight and play with it (maybe even run your scenario above through it) and check to see the difference between starting with a 0 balance, and with the negative starting balance.

**Reply by Keetman** [reply]
You're welcome.  We can all keep playing with it if there are issues.

**Reply by bpaulien** [reply]
So, what category should it go to when I transfer the money to the off budget account?  The category that the debt is in?  Because when I do that, it clears out the debt, and it looks like there's nothing to pay.  Is that what it should be doing?

EDIT: The "old debt" account, that I want to move off budget.  If I put the transaction to category of "old debt" then it clears the old debt.  But I still have that to  pay.  Or is that just going to be a transfer to off budget account, to the category of "old debt" and I budget to the "old debt" each month to pay it off?

**Reply by Keetman** [reply]
I tweaked my fake budget for you a little.  Hopefully this answers your questions.

There are 2 scenarios.  

One you are dealing with a card that you are continuing to use every month (and pay off current charges, but not the old debt).  This situation I've talked about above.  In that situation, you budget an amount that you'd like to pay on the old debt.  If you configure it how I've shown, as soon as you budget that amount, the old debt category will be decreased by that amount.  The credit card balance will not show a decrease until a payment is made.  At some point you will do a transfer from checking to the on budget credit card.  You will not pick a category, just a transfer of the amount you budgeted for "old debt".  The on budget credit card balance will now match the amount budgeted in the "old debt" category.  No category was need for the transfer.  It isn't possible to do so, but you're also talking about old debt so a category isn't even needed.  Any new purchases will be categorized and when you make that payment, the balances on the on budget card and the "old debt" category will still be in sync.  (This is because when you made the purchase on the credit card the CC balance went up, but the "old debt" category did not).


The second scenario is an off budget card that has old debt.  In this situation you simply make another category for old debt (separate from any other on budget debt).  Then you budget an amount for that category.  When it's time for you to make a payment you will still make a transfer.  The transaction will be a transfer to the off budget credit card.  You WILL need a category this time because you are moving money off the budget.  You will use the off budget debt category and you will see the available balance in that category go to zero.  Everything should be balanced at that point.


Here I kept the amount for old debt at $1000.  I just budgeted $800 for the on budget "old debt" and $200 for the off budget old debt.  I have screen shots of the budget and all three accounts.

![image](https://user-images.githubusercontent.com/94420194/142297215-60493d43-cda8-4c44-a9dc-092850adeac1.png)


![image](https://user-images.githubusercontent.com/94420194/142297277-dab98f12-940f-4fe3-a8a5-414d1ffd11a3.png)

![image](https://user-images.githubusercontent.com/94420194/142297344-c8f76f6f-ff14-489b-b01c-859a82966e6e.png)

![image](https://user-images.githubusercontent.com/94420194/142297402-d84229d9-ae12-49c7-b8f1-89fbd0ec235a.png)

**Reply by bpaulien** [reply]
I think I'm getting it.  Arrgh!  I got the off budget one squared away, and think I'm good on the other one.  I'll just keep transferring money and budgeting to the account for the off budget debt until it's gone.

And the on budget one, as long as the negative balance on that category is less than what I owe (riding the CC float) I'll whittle it down too, budgeting various extra $'s here and there to it, until I'm all caught up.

**Reply by Keetman** [reply]
Well, that issue is purely budgeting and finances.  I did that about 13 years ago and I feel your pain.  It is such a relief when the yoke is gone though, so stay motivated!

Hopefully you can figure out the software enough to make it flow well too.  I think once you've got it set up, it shouldn't be too much work.
